Add support for ruff, pre-commit, commitlint, pyproject.toml and dependabot

mhieta opened the pull request:
Updates:
- Use ruff instead of Black, Flake8 and isort
- Use commitlint
- Use pyproject.toml instead of setup.cfg
- Update requirements
- Add pre-commit to project
- Update project readme
- Use common CI setup
- Add dependabot CI-config
- Update release-please configuration

Refs: [RATYK-92](https://helsinkisolutionoffice.atlassian.net/browse/RATYK-92), [RATYK-93](https://helsinkisolutionoffice.atlassian.net/browse/RATYK-93)

[RATYK-92]: https://helsinkisolutionoffice.atlassian.net/browse/RATYK-92?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[RATYK-93]: https://helsinkisolutionoffice.atlassian.net/browse/RATYK-93?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

terovirtanen commented:
**PAIKKATIETOHAKU-API branch is deployed to platta: https://paikkatietohaku-api-pr73.api.dev.hel.ninja** :rocket::rocket::rocket:

danipran commented on .github/workflows/ci.yml:
```diff
@@ -1,51 +1,16 @@
+    secrets: inherit
+    with:
+      python-version: "3.12"
+      postgres-major-version: 13
```
Is this correct? The compose file uses version 16.

mhieta replied:
Updated to use Postgres version 16.

danipran commented on README.md:
```diff
@@ -16,8 +16,8 @@ The project is now running at [localhost:8080](http://localhost:8080)
 
-* PostgreSQL 11 or higher with PostGIS extension
-* Python 3.8 or higher
+* PostgreSQL 13 or higher with PostGIS extension
+* Python 3.9 or higher
```
Are these correct? Compose file specifies PostgreSQL ver. 16 and the CI workflow uses Python 3.12.

mhieta replied:
Updated to use PostgreSQL version 16 and Python 3.12.

danipran commented on pyproject.toml:
```diff
@@ -0,0 +1,36 @@
+[tool.ruff]
+target-version = "py39"
```
Is this correct? CI workflow uses Python 3.12.

mhieta replied:
Updated to use Python 3.12.

danipran commented on requirements-dev.in:
```diff
@@ -1,8 +1,6 @@
 pytest
 pytest-cov
 pytest-django
+ruff==v0.9.10 # sync with .pre-commit-config.yaml
```
Remove the `v` prefix:
```suggestion
ruff==0.9.10 # sync with .pre-commit-config.yaml
```

danipran reviewed (changes requested):
There's some discrepancy with Postgres/Python versions within the project. Make sure they're consistent across the board.

Also noticed that 8e5418b7 (ci: migrate to pyproject.toml) contains commitlint changes as well. Split this to its own commit.

terovirtanen commented:
**PAIKKATIETOHAKU-API branch is deployed to platta: https://paikkatietohaku-api-pr73.api.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**PAIKKATIETOHAKU-API branch is deployed to platta: https://paikkatietohaku-api-pr73.api.dev.hel.ninja** :rocket::rocket::rocket:

mhieta commented:
> There's some discrepancy with Postgres/Python versions within the project. Make sure they're consistent across the board.
 
Updated to use PostgreSQL 14 and Python 3.12 across the project.

> Also noticed that [8e5418b](https://github.com/City-of-Helsinki/geo-search/commit/8e5418b7316ca4e8965829b147c1bee42d0ab51f) (ci: migrate to pyproject.toml) contains commitlint changes as well. Split this to its own commit.

Splitted commitlint into own commit.

danipran reviewed (approved):
LGTM! 👍
